compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEInsertCircularDmas.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/IR/AMDAIEDialect.h"
+#include "iree-amd-aie/IR/AMDAIEOps.h"
+#include "iree-amd-aie/Transforms/Passes.h"
+
+#define DEBUG_TYPE "iree-amdaie-insert-circular-dma"
+
+namespace mlir::iree_compiler::AMDAIE {
+
+namespace {
+
+/// Insert circular dma copy operations from L2 to L1 and L1 to L2. This is
+/// hardcoded for now because of how MLIR-AIE is set up, but can be generalized
+/// in the future.
+LogicalResult insertCircularDmaCpyNdOps(Operation *op) {
```

Comment:
This pass seems to just take the `AMDAIE::DmaCpyNdOp` between L1 and L2 memory spaces and replace it with this new op `AMDAIE::CircularDmaCpyNdOp`. First question, is it a necessary process? In your following passes, whenever you need it can you just directly detect the memory spaces like this pass does?

Second thing, if you have to keep this new op and pass, it sounds more reasonable to me if the name of the pass is `convertToCircularDmaCpyNdOps`. Because what it does is replacing the op instead of inserting a new op.

Reply:
Earlier, I had it in a subsequent pass and I got it out to simplify that one. I am also expecting that this logic can become more complex in the future as:
- You could do L1 -> L2 and L2 -> L1 non-circular as well, but I don't think that works with current MLIR-AIE.
- We will probably have to insert new transformations after this pass and before the next one, for example to hoist these circular buffers  with no dependencies on the surrounding loops out of those loops.  This is a valid transformation for `CircularDmaCpyNdOp` in certain circumstances, but not for `DmaCpyNdOp`:

```
scf.for %i = ... {
  %0 = amdaie.circular_dma_cpy_nd(...)
  ...
}  
```

to 

```
%0 = amdaie.circular_dma_cpy_nd(...)
scf.for %i = ... {
  ...
}  
```

Reply:
> Second thing, if you have to keep this new op and pass, it sounds more reasonable to me if the name of the pass is convertToCircularDmaCpyNdOps. Because what it does is replacing the op instead of inserting a new op.

Good point, thanks. I made this change.